From c977f7da84e16be1fab5942898b5dfea90f43010 Mon Sep 17 00:00:00 2001
Message-Id: <c977f7da84e16be1fab5942898b5dfea90f43010.1665062571.git.keller@fs-net.de>
From: Hartmut Keller <keller@fs-net.de>
Date: Thu, 6 Oct 2022 15:20:21 +0200
Subject: [PATCH 1/2] Fix compilation issues

Fix a strange use of strncpy() which is in fact a strcpy(), but most
probably unintentionally drops the closing zero byte.

Using MAX on two kernel versions seems to fail on gcc. So change this
to a simpler replacement.
---
 mxm_wifiex/wlan_src/mlinux/moal_main.c | 3 +--
 mxm_wifiex/wlan_src/mlinux/moal_main.h | 6 +++++-
 2 files changed, 6 insertions(+), 3 deletions(-)

diff --git a/mxm_wifiex/wlan_src/mlinux/moal_main.c b/mxm_wifiex/wlan_src/mlinux/moal_main.c
index afe35ec..4e32cb5 100644
--- a/mxm_wifiex/wlan_src/mlinux/moal_main.c
+++ b/mxm_wifiex/wlan_src/mlinux/moal_main.c
@@ -9640,8 +9640,7 @@ mlan_status woal_request_country_power_table(moal_private *priv, char *country)
 				       (last_slash - file_path));
 		else
 			memset(file_path, 0, sizeof(file_path));
-		strncpy(file_path + strlen(file_path), country_name,
-			strlen(country_name));
+		strcpy(file_path + strlen(file_path), country_name);
 		PRINTM(MMSG, "Trying again download country_power_tble: %s\n",
 		       file_path);
 		ret = woal_set_user_init_data(handle, COUNTRY_POWER_TABLE,
diff --git a/mxm_wifiex/wlan_src/mlinux/moal_main.h b/mxm_wifiex/wlan_src/mlinux/moal_main.h
index e1b2283..854f741 100644
--- a/mxm_wifiex/wlan_src/mlinux/moal_main.h
+++ b/mxm_wifiex/wlan_src/mlinux/moal_main.h
@@ -160,7 +160,11 @@ Change log:
 #endif
 
 #define COMPAT_VERSION_CODE KERNEL_VERSION(0, 0, 0)
-#define CFG80211_VERSION_CODE MAX(LINUX_VERSION_CODE, COMPAT_VERSION_CODE)
+#if COMPAT_VERSION_CODE > LINUX_VERSION_CODE
+#define CFG80211_VERSION_CODE COMPAT_VERSION_CODE
+#else
+#define CFG80211_VERSION_CODE LINUX_VERSION_CODE
+#endif
 
 /**
  * Reason Code 3: STA is leaving (or has left) IBSS or ESS
-- 
2.14.5

